### PR TITLE
BUG: Dickey Fuller constant values (issue #8471 )

### LINKED ISSUE #8471: DIckey Fuller test breaks on constant values
#### Describe the bug

Dickey-Fuller test gives error when testing with constant values.

#### Code Sample, a copy-pastable example if possible

```python
adfuller(np.full(8, 5.0))
```
`RuntimeWarning: divide by zero encountered in log
  llf = -nobs2*np.log(2*np.pi) - nobs2*np.log(ssr / nobs) - nobs2`

#### Expected Output

I have expected that this series is stationary

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

INSTALLED VERSIONS
------------------
Python: 3.10.6.final.0
OS: Linux 5.15.0-50-generic #56-Ubuntu SMP Tue Sep 20 13:23:26 UTC 2022 x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

statsmodels
===========

Installed: 0.13.2 (/home/michael/.local/lib/python3.10/site-packages/statsmodels)

Required Dependencies
=====================

cython: Not installed
numpy: 1.23.3 (/home/michael/.local/lib/python3.10/site-packages/numpy)
scipy: 1.9.2 (/home/michael/.local/lib/python3.10/site-packages/scipy)
pandas: 1.5.0 (/home/michael/.local/lib/python3.10/site-packages/pandas)
    dateutil: 2.8.2 (/home/michael/.local/lib/python3.10/site-packages/dateutil)
patsy: 0.5.3 (/home/michael/.local/lib/python3.10/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.6.1 (/home/michael/.local/lib/python3.10/site-packages/matplotlib)
    backend: module://matplotlib_inline.backend_inline 
cvxopt: Not installed
joblib: Not installed

Developer Tools
================

IPython: 8.5.0 (/home/michael/.local/lib/python3.10/site-packages/IPython)
    jinja2: 3.1.2 (/home/michael/.local/lib/python3.10/site-packages/jinja2)
sphinx: Not installed
    pygments: 2.13.0 (/home/michael/.local/lib/python3.10/site-packages/pygments)
pytest: Not installed
virtualenv: 20.16.5 (/home/michael/.local/lib/python3.10/site-packages/virtualenv)

</details>

### PR DESCRIPTION
**Pull request by athompson1991**
- [x] closes #8471
- [x] tests added / passed. 
- [x] code/documentation is well formatted.  
- [x] properly formatted commit message.
<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/tsa/stattools.py [hunk 1]
```diff
@@ -268,6 +268,9 @@ def adfuller(
     store = bool_like(store, "store")
     regresults = bool_like(regresults, "regresults")
 
+    if x.max() == x.min():
+        raise ValueError("Invalid input, residual variance is zero")
```
Comment (josef-pkt) [inline statsmodels/tsa/stattools.py:272]: I think this should have a more specific message, e.g.
`ValueError("Invalid data, x is constant")`

Comment (athompson1991) [reply statsmodels/tsa/stattools.py]: Hi, just updated that message to report "x is constant"

**Comment by josef-pkt** [comment]
can you rebase and squash this, then I will merge it.

otherwise, I can do the squashing

**Comment by josef-pkt** [comment]
squashed correctly but not rebased on current main, but that's not a problem in this case

merging

Thanks for the PR

**Pull request merged**